src/profiles.js

```diff
@@ -0,0 +1,199 @@
+import tv4 from 'tv4'
+import jsonlint from 'json-lint'
+import _ from 'lodash'
+import path from 'path'
+
+import Utils from './utils'
+
+const DEFAULT_REMOTE_PATH = 'http://schemas.datapackages.org/registry.csv'
+let DEFAULT_LOCAL_PATH
+if (!Utils.isBrowser) {
+  DEFAULT_LOCAL_PATH = path.join(__dirname, '..', 'schemas', 'registry.csv')
+}
+
+class Profiles {
+
+  constructor(remote = false) {
+    const self = this
+        , PATH = remote || Utils.isBrowser ? DEFAULT_REMOTE_PATH :
+                 DEFAULT_LOCAL_PATH
+
+    return new Promise((resolve, reject) => {
+      const promises = []
+      promises.push(this._loadRegistry(PATH))
+      promises.push(this._loadBasePath(PATH))
+
+      Promise.all(promises).then(values => {
+        self._registry = values[0]
+        self._base_path = values[1]
+        this._getProfiles().then(profiles => {
+          self._allProfiles = profiles
+          resolve(self)
+        }).catch(err => {
+          reject(err)
+        })
+      }).catch(err => {
+        reject(err)
+      })
+    })
+  }
+
+  /**
+   * Retrieve a profile by id.
+   *
+   * @param {String} [profile='base']
+   * @return {Promise}
+   */
+  retrieve(profile = 'base') {
+    return this._allProfiles[profile]
+  }
+
+  /**
+   * Validate a descriptor against a profile. You can provide custom schema
+   * or specify the profile id.
+   *
+   * @param {String|JSON} descriptor The descriptor that needs to be validated
+   * @param {String|JSON} profile Schema to validate against
+   * @return {Promise} Resolves `true` or array of strings which explain the
+   *   errors.
+   */
+  validate(descriptor, profile = 'base') {
+    function _tv4validation(data, schema) {
+      const validate = tv4.validateMultiple(data, schema)
+      if (validate.valid) {
+        return true
+      }
+
+      return Utils.errorsToStringArray(validate.errors)
+    }
+
+    let json = descriptor
+
+    if (typeof descriptor === 'string') {
+      const lint = jsonlint(descriptor)
+      if (lint.error) {
+        return Utils.errorsToStringArray(new Error(lint.error))
+      }
+
+      json = JSON.parse(descriptor)
+    }
+
+    if (_.isObject(profile) && !_.isArray(profile) && !_.isFunction(profile)) {
+      return _tv4validation(json, profile)
+    }
+
+    return _tv4validation(json, this.retrieve(profile))
+  }
+
+  // ------ Private methods  -------
+
+  /**
+   * Returns the base path.
+   *
+   * @return {String}
+   * @private
+   */
+  get _basePath() {
```

Comment:
I can safely remote it, this is only used once by the class and since it's considered a private I think the class can do just fine only with `this._base_path` (and will rename it to `this._basePath` - thanks for the heads up)